script/README.md

```diff
@@ -0,0 +1,155 @@
+# Old scripts deprecation notice
+
+Some of the scripts (and files used by them) in this folder are deprecated and will be moved/reworked in the future. These are:
+
+- `networks-eip-1559` and `networks-legacy` files.
+- Script templates from `utils` folder, and scripts using them.
+- Scripts from `sh` folder.
+
+# Usage of the test script templates.
+
+In this repo we have two types of Foundry test files (`*.t.sol`)
+
+- Base tests: these are named `contract *Test`.
+- Fork tests: these are named `contract *TestFork`.
+
+## Base tests
+
+Base tests are unit or integration tests that either don't require any live state, or that could use the simulated state.
+They do not require any special setup and could be launched using:
+
+```bash
+# Runs all base tests
+$ ./script/foundry-base.sh
+# Runs base tests for contracts having Bus in their name
+$ ./script/foundry-base.sh Bus
+```
+
+Base tests are a part of the CI workflow called `foundry-base`.
+
+## Fork tests
+
+Fork tests are written for contract upgrades and new integrations. These should use the RPC URL from the `.env` file (check `.env.example` for the env variable name).
+
+Fork tests could be launched using a separate script:
+
+```bash
+# Runs all fork tests
+$ ./script/foundry-fork.sh
+# Runs fork tests for contracts having Klaytn in their name
+$ ./script/foundry-fork.sh Klaytn
+```
+
+Fork tests are not a part of the CI workflow. Instead, they are supposed to be run before the deployment goes live.
+
+# Usage of the deployment script templates.
+
+## Setting up the environment variables.
+
+1. Copy the `env.example` file to `.env`.
+2. Replace `PasteYourOwnKeyPlease` with your own etherscan-clone API key for the chains you want to deploy verified contracts.
+3. Feel free to replace some of the RPC urls, if the default ones are not working for you.
+4. Define wallets as per the example in the file.
+   > Available options:
+   >
+   > - Keystore files, currently foundry only supports keystore files saved by `cast wallet new <path>` command, keystores exported from other sources will not work.
+   > - Ledger and Trezor wallets.
+   > - Interactive prompt for entering the private key.
+5. Make sure you have enough ETH in the wallets to pay for the gas fees.
+
+## Available chains for deployment.
+
+Every chain has its own name, which is used in `.env`, `foundry.toml`, `deployments` and `script/configs`. Therefore, it is important to use the same name for the chain everywhere. The list of chains could be found in `foundry.toml` file.
```

Comment:
Would link to each of these.

Reply:
Done in 58bb0a3d308dfeaea121f53f160686a45b06726c